gems/smithy/lib/smithy/anvil/client/views/endpoint_provider.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+require 'stringio'
+
+module Smithy
+  module Anvil
+    module Client
+      module Views
+        # @api private
+        class EndpointProvider < View
+          def initialize(plan)
+            @plan = plan
+            @model = plan.model
+            service = Vise::ServiceIndex.new(@model).service.values.first
```

Comment:
Outside scope, I am thinking we can put a resolved service id on the plan, so fetching the service shape would be to index into shapes with it. There wouldn't be an index I don't think. That would also force validation instead of first index usage.